### PR TITLE
add example

### LINKED ISSUE #13: Add simple binary example
We should have an example in this repo similar to `deno_lint`'s example binary called [`dlint`](https://github.com/denoland/deno_lint/tree/master/examples/dlint).

### PR DESCRIPTION
**Pull request by zhmushan**
close #13

### CODE REVIEW

File: examples/ddoc/main.rs [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    let fetcher = self.clone();
+    let module_url = Url::parse(specifier).unwrap();
+    async move {
+      let url_scheme = module_url.scheme();
+      let is_local_file = url_scheme == "file";
+
+      if is_local_file {
+        Ok(fetcher.fetch_local_file(&module_url))
+      } else {
+        Ok(fetcher.fetch_remote_source(&module_url))
```
Comment (bartlomieju) [inline examples/ddoc/main.rs:57]: Let's not add handling for remote sources at all - please return error when user tries to load remote source

File: examples/ddoc/main.rs [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+impl SourceFileFetcher {
+  pub fn new() -> Box<Self> {
+    Box::new(Self {})
+  }
+
+  fn fetch_local_file(&self, module_url: &Url) -> String {
+    let path = module_url.to_file_path().unwrap();
+    read_to_string(path).unwrap()
+  }
```
Comment (bartlomieju) [inline examples/ddoc/main.rs:22]: This impl block is superfluous

File: examples/ddoc/main.rs [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+        Ok(fetcher.fetch_remote_source(&module_url))
+      }
+    }
+    .boxed_local()
+  }
+}
+
+fn main() {
+  let args: Vec<String> = args().collect();
+  let target = args.get(1).unwrap();
```
Comment (bartlomieju) [inline examples/ddoc/main.rs:66]: Maybe add `clap` for nicer CLI?

**Review by bartlomieju — CHANGES_REQUESTED** [review]

**Comment by zhmushan** [comment]
@bartlomieju done

File: Cargo.toml [hunk 4]
```diff
@@ -18,5 +24,6 @@ termcolor = "1.1.0"
 regex = "1.3.9"
 
 [dev-dependencies]
+clap = "2.33.3"
```
Comment (ry) [inline Cargo.toml:27]: It’s just another thing to keep in-sync. I’d think it’d be better if the example was just a minimal example of integrating with deno_doc

Comment (lucacasonato) [reply Cargo.toml:27]: This is a dev dependency. It doesn't need to be kept in sync. Just like for `deno_lint`.

Comment (zhmushan) [reply Cargo.toml:27]: This is my first time using Clap, and it is indeed a good start for creating cli applications.  But I have no idea whether to recommend it to others as an example. This is a trivial matter and users will not be confused by it.

**Review by bartlomieju — APPROVED** [review]
LGTM, thanks @zhmushan

**Pull request merged**